src/proxy/inbound.rs

```diff
@@ -514,9 +516,12 @@ impl Inbound {
         Ok(())
     }
 
-    /// find_inbound_upstream determines the next hop for an inbound request.
-    #[expect(clippy::type_complexity)]
-    fn find_inbound_upstream(
+    /// Resolves the target address for an inbound request.
+    /// This handles both regular workload traffic and ztunnel's own services.
+    /// 
+    /// When ztunnel is the target workload (identified by workload name starting with "ztunnel"),
+    /// this function will handle routing to local services like metrics or metadata endpoints.
+    pub(super) fn resolve_inbound_target(
         cfg: &Config,
```

Comment:
All the other changes were backed out, we should probably also go ahead and back out the rename and these comments which no longer apply.